pkgs/development/compilers/haxe/default.nix

```diff
@@ -1,14 +1,38 @@
-{ lib, stdenv, fetchgit, coreutils, ocamlPackages, zlib, pcre, neko }:
-
-let inherit (ocamlPackages) ocaml camlp4; in
+{ lib, stdenv, fetchgit, coreutils, ocaml-ng, zlib, pcre, neko, mbedtls }:
 
 let
-  generic = { version, sha256, prePatch }:
+  ocamlDependencies = version:
+    if lib.versionAtLeast version "4.0"
+    then with ocaml-ng.ocamlPackages; [
+      ocaml
+      findlib
+      sedlex_2
+      xml-light
+      ptmap
+      camlp5
+      sha
+      dune_2
+      luv
+    ] ++ (if lib.versionAtLeast version "4.2" then [
```

Comment:
do you really want to maintain those `if`/`then` when next Haxe version is out ?
inlining `ocamlDependencies` function removes them

Reply:
Sure, why not. I agree it's a bit ugly, but when extlib 1.7.9 comes out we'll be able to get rid of this.

Reply:
I can predict what will be: at some point you will be so unhappy with similar contributions by others that you fork out and freeze `nixpkgs` for personal use (probably undoing there most of own contributions), and no one will maintain the ruins in `master`.

Reply:
Not sure what you getting at, I generally maintain my contributions and have no intention of maintaining a fork of nixpkgs because why would I?

My intention with this function is merely to reduce code duplication, so we don't need to make the same adjustment in 10 places in the future if for example something in `ocamlPackages` changes.

Reply:
Rebasing onto #117113 allows us to get rid of the extlib conditional already today.